File: framework/languages/java/samples/java/ShoppingMall/Client/src/main/java/systems/zlink/samples/shoppingmall/client/configuration/SampleNames.java
package systems.zlink.samples.shoppingmall.client.configuration;

public final class SampleNames {
    public static final String CommerceApiChannel = "shoppingmall.commerce.api";
    public static final String ApiInstanceA = "api-a";
    public static final String ApiInstanceB = "api-b";

    private SampleNames() {
    }

    public static String commerceApiChannel(String instanceId) {
        return CommerceApiChannel + "." + instanceId;
    }
}

File: framework/languages/java/samples/kotlin/ShoppingMallCheckout/Server/Configuration/src/main/kotlin/systems/zlink/samples/kotlin/shoppingmallcheckout/server/configuration/SampleNames.kt
package systems.zlink.samples.kotlin.shoppingmallcheckout.server.configuration

object SampleNames {
    const val CommerceApiChannel = "shoppingmall.commerce.api"
    const val OrderWorkflowChannel = "shoppingmall.order.workflow"

    const val ApiInstanceA = "api-a"
    const val ApiInstanceB = "api-b"
    const val WorkflowInstanceA = "workflow-a"
    const val WorkflowInstanceB = "workflow-b"

    fun workflowChannel(instanceId: String): String = "$OrderWorkflowChannel.$instanceId"

    fun commerceApiChannel(instanceId: String): String = "$CommerceApiChannel.$instanceId"
}
